lib/active_merchant/billing/gateways/monei.rb

```diff
@@ -0,0 +1,314 @@
+            xml.Amount amount(money)
+            xml.Currency options[:currency] || currency(money)
+            xml.Usage options[:description] || options[:order_id]
+          end unless money.nil?
+        end
+      end
+
+      # Private: Add account part to XML
+      def add_account(xml, credit_card)
+        xml.Account do
+          xml.Holder credit_card.name
+          xml.Number credit_card.number
+          xml.Brand credit_card.brand.upcase
+          xml.Expiry(:month => credit_card.month, :year => credit_card.year)
+          xml.Verification credit_card.verification_value
+        end
+      end
+
+      # Private: Add customer part to XML
+      def add_customer(xml, credit_card, options)
+        requires!(options, :billing_address)
+        address = options[:billing_address]
+        xml.Customer do
+
+          xml.Name do
+            xml.Given credit_card.first_name
+            xml.Family credit_card.last_name
+          end
+          xml.Address do
+            xml.Street address[:address1].to_s
+            xml.Zip address[:zip].to_s
+            xml.City address[:city].to_s
+            xml.State address[:state].to_s if address.has_key? :state
+            xml.Country address[:country].to_s
+          end
+          xml.Contact do
+            xml.Email options[:email] || 'noemail@monei.net'
+            xml.Ip options[:ip] || '0.0.0.0'
+          end
+
+        end
+      end
+
+      # Private: Parse XML response from Monei servers
+      def parse(body)
+        xml = Nokogiri::XML(body)
+
+        {
+            :unique_id => xml.xpath("//Response/Transaction/Identification/UniqueID").text,
+            :status => translate_status_code(xml.xpath("//Response/Transaction/Processing/Status/@code").text),
+            :reason => translate_status_code(xml.xpath("//Response/Transaction/Processing/Reason/@code").text),
+            :message => xml.xpath("//Response/Transaction/Processing/Return").text
+        }
+      end
+
+      # Private: Send XML transaction to Monei servers and create AM response
+      def commit(xml)
+        url = (test? ? test_url : live_url)
+
+        response = parse(ssl_post(url, post_data(xml), 'Content-Type' => 'application/x-www-form-urlencoded;charset=UTF-8'))
+
+        Response.new(
+          success_from(response),
+          message_from(response),
+          response,
+          authorization: authorization_from(response),
+          test: test?,
+          error_code: error_code_from(response)
+        )
+      end
+
+      # Private: Decide success from servers response
+      def success_from(response)
+        response[:status] == :success || response[:status] == :new
+      end
+
+      # Private: Get message from servers response
+      def message_from(response)
+        response[:message]
+      end
+
+      # Private: Get error code from servers response
+      def error_code_from(response)
+        success_from(response) ? nil : STANDARD_ERROR_CODE[:card_declined]
```

Comment:
Hi @j-mutter,

The only codes that we get from the API are:

NEW 90
WAITING 80
REJECTED_VALIDATION 70
REJECTED_RISK ￼65
￼REJECTED_BANK 60
￼NEUTRAL ￼40
￼SUCCESS 00

I translated all the rejections to card_declined because I do not get the level of detail in the response that STANDARD_ERROR_CODE has. What do you suggest?